projects/ui/src/Components/Structure/HeaderSectionLoggedIn.tsx

```diff
@@ -1,23 +1,31 @@
 import { Popover } from "@mantine/core";
-import { useContext, useMemo, useState } from "react";
+import { useEffect, useMemo, useState } from "react";
 import { di } from "react-magnetic-di";
-import { NavLink, useLocation } from "react-router-dom";
+import { NavLink, useLocation, useSearchParams } from "react-router-dom";
 import { useGetCurrentUser } from "../../Apis/hooks";
 import { Icon } from "../../Assets/Icons";
-import { PortalAuthContext } from "../../Context/PortalAuthContext";
+import { clientId, logoutEndpoint } from "../../user_variables.tmplr";
 
 const HeaderSectionLoggedIn = () => {
   di(useGetCurrentUser);
   const { data: user } = useGetCurrentUser();
   const routerLocation = useLocation();
-  const { onLogout } = useContext(PortalAuthContext);
   const [opened, setOpened] = useState(false);
 
   const inUsagePlansArea = useMemo(
     () => routerLocation.pathname.includes("/usage-plans"),
     [routerLocation.pathname]
   );
 
+  const [searchParams, setSearchParams] = useSearchParams();
+  useEffect(() => {
+    const newSearchParams = new URLSearchParams(searchParams);
+    newSearchParams.delete("state");
+    newSearchParams.delete("session_state");
+    newSearchParams.delete("code");
+    setSearchParams(newSearchParams);
+  }, [setSearchParams]);
```

Comment:
Is there a reason setSearchParams would be changing, or is this only to do on component load after the setSearchParam is settled?

Honestly, this part is nicely terse, but there's no context nearby for what's happening and its a bit hard to follow for why we are doing this/what the goal of this piece of code is.

Reply:
`setSearchParams` is just in the dependency array for completeness, it shouldn't change.

I didn't add any explanations here, but this basically cleans up the URL after it is redirected from the login screen since it has all that info in it.